### ISSUE TITLE
Weird problem here

### ISSUE THREAD

**Issue by staritbusfans** [takatost/homebridge-mi-ac-partner#2]
I'm using the AC Partnerv2 and having a weird problem here. I discovered the token of it by miio and successfully added it into HomeKit. I can see the response in the Mi Home app while changing the temp and mode setting in Homekit but the AC itself isn't response. It seems that the Partner itself did received the command but it didn't send it out via IR. Controlling in the Mi Home app is normal and I have tried resetting and re-configured it several times.

**Comment by staritbusfans** [comment]
I'm using a GREE's AC.
P.S. Is it able to support sub-accessories such as Temp-Hum sensor under the AC Partner?

**Comment by Dengor1984** [comment]
I have the same problem...

**Comment by mulua6** [comment]
I have the same problem

**Comment by cxlwill** [comment]
Me as well. I checked another issue where author said the reason is that Gree's cmd code is different with preset one, which is Media. Furthermore, to get the correct cmd code needs complex procedure. So I think we can just wait for the update and fresh commit. Btw, thanks for the plugin.

**Comment by takatost** [comment]
@staritbusfans @Dengor1984 @mulua6 @cxlwill Would you plz tell what is your AC brand and preset number in AC Partner settings of Mi Home App?

**Comment by cxlwill** [comment]
Of course. Mine is Gree, so preset number is first one. 我用格力所以应该是第一个。好像最新的 Bluestacks 没法运行在 macOS 10.12 上。所以我也没办法手动提取了。

**Comment by staritbusfans** [comment]
The eighth one and the tenth one are both working in the Mii home App but neither of them works here.

**Comment by takatost** [comment]
@cxlwill 请更新版本至 2.0.0，并且在 homebridge 的 config.json 中增加两行配置 `brand` 和 `preset_no`

范例：
```
     "accessories": [
       {
          "accessory": "MiAcPartner",
          "token": "eda2570a242a2e9f862b7d4ff1c93631",
          "name": "Ac Partner",
          "brand": "gree",
          "preset_no": "1"
       }
     ]
```

然后启动 `homebridge -D`，麻烦测试一下
目前仅添加了两个空调预设，格力和美的的第一个

**Comment by takatost** [comment]
@staritbusfans I added your gree no.8 preset in version 2.0.1

**Comment by cxlwill** [comment]
完美~ 模式选择和温度控制都可以~~由于我的空调是单冷，所以没有办法制热，但是个人认为制热模式也是有效的。格力大部分空调使用的是模式一，其实是一个通用遥控器，所以现在已经解决了很多设备了~ 非常感谢。

**Comment by Dengor1984** [comment]
My Air Conditioner is Cooper&Hunter, but in MiHome app it worcs fine with Gree preset No1

**Comment by cxlwill** [comment]
@Dengor1984 Then I think it is worth to try the latest version of this plugin for you. You can update it and add brand and preset value in config.json, then restart HB. If you work, it is nice to report back~ :)

**Comment by takatost** [comment]
@cxlwill 格力预设模式 1 的命令码相对比较简单，可以看出规律，其他的空调模式可能会有很多不确定性，无法找出规律，从空调开关，制冷、制热、自动，还有 17-30 摄氏度设置，一共有 3×14+1，58 种命令码，没有规律的命令码特别头疼。
之后会陆续加上其他品牌的空调的支持，如果各位有能力，也可以一起抓包给我 pr，谢谢。

**Comment by staritbusfans** [comment]
非常感谢，正在更新

**Comment by cxlwill** [comment]
好的，我会动员其他人帮助抓包~ 发现一个问题，我猜是HB 的局限，改变温度会默认设置为强风模式，还需要进一步手动修改。

**Comment by takatost** [comment]
风速我会研究下恒温器是否支持，当前我设置的默认都是自动风速，如果支持的话我会在后续版本加上该功能，当然工作量也会成指数形式增加。。

**Comment by Dengor1984** [comment]
@takatost i have update the plugin to v2.0.1, change the config to gree No1, have the preset Gree No1 in MiHome.
In homekit Ac Companion shows, but it not active and dont send ir command...

**Comment by Dengor1984** [comment]
And I have installed a Mi-Aqara plugin, maybe this is a problem?

**Comment by staritbusfans** [comment]
现在能开不能关…格力8号

**Comment by EabeyTong** [comment]
Media, No:KFR-35G/BP3DN1Y-YA100(B1)
MI-AC No：KTBL02LM
preset_no: 1，Don't work！

**Comment by maxmib** [comment]
我的设备是1代空调网关，米家应用里的设置是格力1号，config里也是设置的格力1号，但是homekit里控制不了空调，开关和调温度都没有反应，用miio discover命令查看设备时网关model是unknown，看ac partner插件运行的log，进行温度控制时会显示device not exist,是因为miio不能识别网关的原因吗

**Comment by LASER-Yi** [comment]
@takatost 我的米家中设置的是格力2号，可以帮忙抓包，想问一下怎么发送抓包信息给你

**Comment by takatost** [comment]
@maxmib 一般情况是 token 有误

**Comment by takatost** [comment]
@LASER-Yi 这篇文章能帮助到你 https://github.com/aholstenson/miio/blob/master/docs/protocol.md
需要的是开关空调、17-30 度各模式的代码

最近比较忙，顾不太上插件

**Comment by LASER-Yi** [comment]
这个无法控制的问题应该是miio没有正确识别空调伴侣的Model，部分miio可能因为序号的原因识别不出来，需要修改你们的miio或者作者使用另一种方式来识别空调伴侣（比如通过ip或者mac地址）

**Comment by maxmib** [comment]
@LASER-Yi 有修改miio 的相关资料可以参考吗

**Comment by penbox** [comment]
我尝试添加两个Ac Partener的时候，只能有一个生效。对另外一个Ac Partener操作的时候会提示“Device not exists.”

**Comment by LASER-Yi** [comment]
我改了一些作者的代码，加了通过ip进行连接的方式和自定义IR信号以用来方便没有预设但会抓包的人使用，[Github地址](https://github.com/LASER-Yi/homebridge-mi-acpartner)

**Comment by newsam3** [comment]
Is the ac partner2 possible to work like as power outlet plug that just work like the zigbee power outlet?
AC partner2 可以像zigbee電源插座一樣只控制電源開閉嗎？
